### PR TITLE
fix(ingest/bigquery): show report in output

### PR DESCRIPTION
**Pull request by hsheth2**
Follow up on https://github.com/datahub-project/datahub/pull/8626.

The report currently shows the object refs:

```
 'schema_api_perf': '<datahub.ingestion.source.bigquery_v2.bigquery_report.BigQuerySchemaApiPerfReport object at 0x7f27859c5ba0>',
 'audit_log_api_perf': '<datahub.ingestion.source.bigquery_v2.bigquery_report.BigQueryAuditLogApiPerfReport object at 0x7f27859c5bd0>',
```

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py [hunk 1]
```diff
@@ -16,7 +17,7 @@
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-class BigQuerySchemaApiPerfReport:
+class BigQuerySchemaApiPerfReport(Report):
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:20]:
```suggestion
@dataclass
class BigQuerySchemaApiPerfReport(Report):
```

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py [hunk 2]
```diff
@@ -25,7 +26,7 @@ class BigQuerySchemaApiPerfReport:
     get_views_for_dataset = PerfTimer()
 
 
-class BigQueryAuditLogApiPerfReport:
+class BigQueryAuditLogApiPerfReport(Report):
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:29]:
```suggestion
@dataclass
class BigQueryAuditLogApiPerfReport(Report):
```

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py [hunk 3]
```diff
@@ -16,7 +17,7 @@
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-class BigQuerySchemaApiPerfReport:
+class BigQuerySchemaApiPerfReport(Report):
     list_projects = PerfTimer()
     list_datasets = PerfTimer()
     get_columns_for_dataset = PerfTimer()
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:23]:
```suggestion
    list_projects: PerfTimer = field(default_factory=PerfTimer)
    list_datasets: PerfTimer = field(default_factory=PerfTimer)
    get_columns_for_dataset: PerfTimer = field(default_factory=PerfTimer)
```

Not able to add suggestions for all here, but this is required for all fields in this and next report.

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py [hunk 4]
```diff
@@ -25,7 +26,7 @@ class BigQuerySchemaApiPerfReport:
     get_views_for_dataset = PerfTimer()
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:26]:
```suggestion
    get_views_for_dataset: PerfTimer = field(default_factory=PerfTimer)
```

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py [hunk 5]
```diff
@@ -25,7 +26,7 @@ class BigQuerySchemaApiPerfReport:
...
 
 
-class BigQueryAuditLogApiPerfReport:
+class BigQueryAuditLogApiPerfReport(Report):
     get_exported_log_entries = PerfTimer()
     list_log_entries = PerfTimer()
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:31]:
```suggestion
    get_exported_log_entries: PerfTimer = field(default_factory=PerfTimer)
    list_log_entries: PerfTimer = field(default_factory=PerfTimer)
```

**Review by mayurinehate — CHANGES_REQUESTED** [review]
Oops. Thanks for catching this. This did work earlier when using inheritance, and stopped working after refractoring to use composition.

I quickly tried these changes and it still didn't work on my machine. Only printed empty objects

```
 'schema_api_perf': {},
 'audit_log_api_perf': {},
```

With suggested changes, it works as expected.

**Comment by hsheth2** [comment]
@mayurinehate thanks - made those changes

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py [hunk 6]
```diff
@@ -16,18 +17,20 @@
...
-    list_projects = PerfTimer()
-    list_datasets = PerfTimer()
-    get_columns_for_dataset = PerfTimer()
+@dataclass
+class BigQuerySchemaApiPerfReport(Report):
+    list_projects: PerfTimer = field(default_factory=PerfTimer)
+    list_datasets: PerfTimer = field(default_factory=PerfTimer)
+    get_columns_for_dataset: PerfTimer = field(default_factory=PerfTimer)
     get_tables_for_dataset = PerfTimer()
     list_tables = PerfTimer()
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:26]:
```suggestion
    get_tables_for_dataset: PerfTimer = field(default_factory=PerfTimer)
    list_tables: PerfTimer = field(default_factory=PerfTimer)
```
Now, all timers are covered :) .

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_report.py:26]: @hsheth2 Can you please make this change also ?

**Review by asikowitz — APPROVED** [review]

**Pull request merged**